lib/workload/stateless/stacks/metadata-manager/app/models/base.py

```diff
@@ -42,7 +43,9 @@ def reduce_multi_values_qor(key: str, values: List[str]):
         ):
             values = [values]
         return reduce(
-            operator.or_, (Q(**{"%s__iexact" % key: value})
+            # Apparently the `get_prep_value` from the custom fields.py is not called prior hitting the Db but,
+            # the regular `__exact` still execute that function.
+            operator.or_, (Q(**{"%s__exact" % key: value})
```

Comment:
One caveat I found here is the strip/annotation will not work if it uses `iexact` on the filtering based on the field model. Currently changing the filter to only match for an exact match.